ingest.py

```diff
@@ -83,60 +84,89 @@ def load(self) -> List[Document]:
 
 def load_single_document(file_path: str) -> Document:
     ext = "." + file_path.rsplit(".", 1)[-1]
-    if ext in LOADER_MAPPING:
-        loader_class, loader_args = LOADER_MAPPING[ext]
-        loader = loader_class(file_path, **loader_args)
-        return loader.load()[0]
 
-    raise ValueError(f"Unsupported file extension '{ext}'")
+    loader_class, loader_args = LOADER_MAPPING[ext]
+    loader = loader_class(file_path, **loader_args)
+    return loader.load()[0]
 
 
-def load_documents(source_dir: str, ignored_files: List[str] = []) -> List[Document]:
+def load_documents(
+    source_dir: str,
+    ingested_files: Optional[List[str]] = None,
+    report_processed_files: bool = False,
+    report_skipped_files: bool = True,
+    report_previously_ingested_files: bool = False,
+    use_process_bar: bool = True,
+) -> List[Document]:
+
     """
     Loads all documents from the source documents directory, ignoring specified files
     """
-    all_files = []
-    for ext in LOADER_MAPPING:
-        all_files.extend(
-            glob.glob(os.path.join(source_dir, f"**/*{ext}"), recursive=True)
-        )
-    filtered_files = [file_path for file_path in all_files if file_path not in ignored_files]
+    if ingested_files is None:
+        ingested_files : List[str] = []
+
+    filtered_files = []
+    extensions = tuple(LOADER_MAPPING.keys())
+
+    for root, _, files in os.walk(source_dir):
+        for file in files:
+            if file.endswith(extensions):
+                file_path = os.path.join(root, file)
+                if file_path not in ingested_files:
+                    filtered_files.append(file_path)
+                else:
+                    if report_ingested_files:
```

Comment:
Typo here, `report_ingested_files` is undefined, you called it `report_previously_ingested_files`.

BTW, sorry if it wasn't clear, I meant clarify to the user. E.g., instead of `(ignore list)` something like `(already ingested)`?

Reply:
1. I forgot to rename both locations.
2. I updated the message to the user.